src/Surfnet/Stepup/Identity/Value/YubikeyPublicId.php

```diff
@@ -49,8 +55,13 @@ public function __toString()
         return $this->value;
     }
 
-    public function equals(YubikeyPublicId $other)
+    public function equals(SecondFactorIdentifier $other)
+    {
+        return $other instanceof self && $this->value === $other->value;
+    }
+
+    public function jsonSerialize()
     {
-        return $this->value === $other->value;
+        return (string) $this;
```

Comment:
for most `jsonSerialize()` I'd advise against using the `__toString()` method, gives finer grained control (they are coincidentally the same, not by design).